### PR TITLE
Inject authz denial messages

### PR DESCRIPTION
**Pull request by n1v0lg**
This PR moves authorization denied message creation from static methods to an injectable class.

Relates: ES-7984

### CODE REVIEW

**Comment by n1v0lg** [comment]
@elasticmachine update branch

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationDenialMessages.java [hunk 1]
```diff
@@ -29,158 +29,202 @@
...
-            + "]";
-
-        return actionIsUnauthorizedMessage
-            + rolesDescription(authentication.getAuthenticatingSubject(), authorizationInfo.getAuthenticatedUserAuthorizationInfo())
-            + ", "
-            + unauthorizedToRunAsMessage;
-    }
-
-    static String actionDenied(
+public interface AuthorizationDenialMessages {
```
[the remark this answers is not in the document]
Comment (slobodanadamovic) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationDenialMessages.java:32]: No worries. It looks much smaller after hiding the whitespaces.

**Review by slobodanadamovic — APPROVED** [review]
LGTM 👍 

I'm completely fine with the inheritance approach here. It follows the same pattern with other API overrides we have. And it might even be better suited if we need to customize other authz denial messages. But just as a thought, the alternative approach could be to have a single implementation and use composition which accepts cluster and index privilege resolvers. This approach might be simpler to reason about (given that there is single `AuthorizationDenialMessages` instance) and have a better separation of concerns, but it's hard to predict the future requirements - so I'm content with inheritance here.

**Comment by n1v0lg** [comment]
>  But just as a thought, the alternative approach could be to have a single implementation and use composition which accepts cluster and index privilege resolvers. 

Absolutely, I had also considered this -- I ultimately decided against it since it feels like a very ad-hoc and context-less thing to inject, whereas AuthzDenialMessages is a reasonable, coherent chunk of functionality. As you say, it also gives us the ability to further customize authz messages down the line, if we choose to.

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**